hera_pspec/pspecdata.py

```diff
@@ -514,12 +517,32 @@ def C_model(self, key, model='empirical'):
         # check cache
         if not self._C.has_key(Ckey):
             # calculate covariance model
+            if model == 'time_average':
+                spw = slice(self.spw_range[0], self.spw_range[1])
```

Comment:
Let's use the objects `self.x()` method to get this data, instead of doing it yourself. That's what this method is for.

Also I don't see you weight the data before taking the time-average: you don't want flagged data in the model I presume. Try using the `self.w()` method to get the data weights and instead of `np.average`, try something like
```
np.sum(d * w, axis=0) / np.sum(w, axis=0).clip(1e-10, np.inf)
```
to create the weighted average.